### ISSUE TITLE
Backslash sometimes seen as escape character in string literal discriminators

### ISSUE THREAD

**Issue by mindplay-dk** [Microsoft/TypeScript#19643]
Try this in the Playground or VSCode:

```ts
interface Foo {
    d: "bar\one";
    data: {
        a: string;
    }
}

interface Bar {
    d: "bar\two";
    data: {
        b: string;
    }
}

declare var test: Foo | Bar;

if (test.d === "bar\one") { 
    test.data.a = "a";
}

if (test.d === "bar\two") { 
    test.data.b = "b";
}
```

Now hover over the `d` in one of the last statements - the Playground displays this:

![image](https://user-images.githubusercontent.com/103348/32274123-be3472d8-bf05-11e7-9ae6-5f9b23502064.png)

Oddly, only in the first of the two discriminators, the `\` is interpreted as an escape code.

Most likely, the text in that popup needs to be escaped, in case it contains backslashes.

**Comment by andy-ms** [comment]
`\o` is not a valid escape but `\t` is -- that escapes a tab.
```sh
$ node
> "bar\one"
'barone'
> "bar\two"
'bar\two'
```

**Comment by mindplay-dk** [comment]
Yeah, since it didn't complain, I wasn't sure if escaping was necessary in type-declarations or not, but I guess it follows the rules of normal string syntax.

Okay, so I guess close this issue, which could be a feature request for the linter maybe. (I'll leave the decision on whether to close it to you.)

Thanks!
